### PR TITLE
Add banner and badge component

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

File: src/atoms/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import type { FunctionComponent, PropsWithChildren } from 'react';
+import { useTheme } from 'styled-components';
+
+import { Body, Box } from '.';
+import type { BannerType } from './Banner';
+
+interface BadgeProps {
```
Comment (Mrtenz) [inline src/atoms/Badge.tsx:7]: Can you export this

File: src/atoms/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  type: BannerType;
+}
+
+export const Badge: FunctionComponent<BadgeProps> = ({
+  type,
+  children
+}: PropsWithChildren<Pick<BadgeProps, 'type'>>) => {
```
Comment (Mrtenz) [inline src/atoms/Badge.tsx:14]: Why?

File: src/atoms/Banner.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+const icons: { [key in BannerType]: string } = {
+  success,
+  info,
+  action,
+  warning,
+  error,
+  clear: action
+};
+
+export const Banner: FunctionComponent<BannerProps> = ({
```
Comment (Mrtenz) [inline src/atoms/Banner.tsx:31]: This should probably be a molecule

File: src/atoms/Banner.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  type,
+  label,
+  badge,
+  extended,
+  children
+}: PropsWithChildren<BannerProps>) => {
```
Comment (Mrtenz) [inline src/atoms/Banner.tsx:37]: Why?

File: src/atoms/Banner.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        px={type === 'clear' && '0'}
+        justifyContent="space-between"
+        onClick={children && handleToggle}
+        sx={{ cursor: children && 'pointer' }}
+        data-testid="banner-toggle">
+        <Flex variant="horizontal-start">
+          <Image src={icons[type]} alt="type" width="20px" minWidth="20px" height="20px" mr="2" />
+          <Body
+            color="inherit"
+            fontSize="12px"
```
Comment (Mrtenz) [inline src/atoms/Banner.tsx:56]: If possible use one of the font sizes defined in the theme. Not sure if they match up

File: src/atoms/Banner.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+              ml="2"
+              width="10px"
+              minWidth="10px"
+              sx={{ userSelect: 'none', transform: isExtended && 'rotate(180deg)' }}
+            />
+          )}
+        </Flex>
+      </Flex>
+      {children && isExtended && (
+        <Body fontSize="14px" p="2" pt="0">
```
Comment (Mrtenz) [inline src/atoms/Banner.tsx:77]: If possible use one of the font sizes defined in the theme. Not sure if they match up

File: src/types/vendor/styled-components.d.ts [hunk 7]
```diff
@@ -1,9 +1,7 @@
...
+import type { Theme } from '../../theme';
 
 declare module 'styled-components' {
-  interface ThemeValue {
-    [key: string]: string | string[] | number | number[] | ThemeValue;
-  }
-
-  export type DefaultTheme = ThemeValue;
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  export interface DefaultTheme extends Theme {}
```
Comment (Mrtenz) [inline src/types/vendor/styled-components.d.ts:6]:
```suggestion
  export type DefaultTheme = Theme;
```
?

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: src/types/vendor/styled-components.d.ts (hunk 7 above)
Comment (FrederikBolding) [reply src/types/vendor/styled-components.d.ts:6]: I tried that initially, but it doesn't work.

File: src/atoms/Badge.tsx (hunk 1 above)
Comment (FrederikBolding) [reply src/atoms/Badge.tsx:7]: Done

File: src/atoms/Banner.tsx (hunk 5 above)
Comment (FrederikBolding) [reply src/atoms/Banner.tsx:56]: Done

File: src/atoms/Banner.tsx (hunk 6 above)
Comment (FrederikBolding) [reply src/atoms/Banner.tsx:77]: It doesn't but I changed it to be `rem`

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**